custom_components/opnsense/update.py

```
@@ -263,71 +204,90 @@
+            await asyncio.sleep(sleep_time)
             try:
-                response = client.upgrade_status()
+                response = self._client.upgrade_status()
                 # after finished status is "done"
-                running = response["status"] == "running"
-            except:
+                running: bool = response["status"] == "running"
+            except Exception as e:
+                exceptions += 1
+                _LOGGER.debug(
+                    f"Error #{exceptions} while getting upgrade_status. {e.__class__.__qualname__}: {e}"
```

Ruff

Logging statement uses f-string

(G004)